lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java

```diff
@@ -3096,6 +3097,19 @@ public int compare(SimpleStudentPage o1, SimpleStudentPage o2) {
 						String folderName = folderPath[folderPath.length-1];
 						if (dataDirectory.endsWith("//")){
 							folderName = simplePageBean.getCurrentSite().getTitle();
+						} else {
+							try {
+								ContentHostingService contentHostingService = (ContentHostingService) ComponentManager.get("org.sakaiproject.content.api.ContentHostingService");
+								ContentCollection collection = contentHostingService.getCollection(dataDirectory.replace("//", "/"));
+								if (collection != null) {
+									String displayName = collection.getProperties().getProperty(collection.getProperties().getNamePropDisplayName());
+									if (displayName != null && displayName != "") {
```

Comment:
Why not just use ResourceProperties.PROP_DISPLAY_NAME?
Also, you could use StringUtils.isBlank  on the displayName test.

Reply:
fixed